### PR TITLE
Expose conformance statement without authentication

### PR DESCRIPTION
**Pull request by VaishSiddharth**
## Issue I worked on

see https://issues.openmrs.org/browse/FM2-

## Checklist: I completed these to help reviewers :)

- [ ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: omod/src/main/java/org/openmrs/module/fhir2/web/filter/AuthenticationFilter.java [hunk 1]
```diff
@@ -40,27 +40,29 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
...
-					// this is "Basic ${base64encode(username + ":" + password)}"
-					try {
-						basicAuth = basicAuth.substring(6); // remove the leading "Basic "
-						String decoded = new String(Base64.decodeBase64(basicAuth), StandardCharsets.UTF_8);
-						String[] userAndPass = decoded.split(":");
-						Context.authenticate(userAndPass[0], userAndPass[1]);
-					}
-					catch (Exception e) {
+			if (!(httpRequest.getRequestURI().contains("/.well-known")
+			        || httpRequest.getRequestURI().endsWith("/metadata"))) {
```
Comment (ibacher) [inline omod/src/main/java/org/openmrs/module/fhir2/web/filter/AuthenticationFilter.java:44]:
```suggestion
			if (!(httpRequest.getRequestURI().contains("/.well-known") || 
			         httpRequest.getRequestURI().endsWith("/metadata"))) {
```

**Comment by VaishSiddharth** [comment]
@ibacher  I don't think this PR solves the errors I have sent you the issues on TALK please have a look

**Comment by VaishSiddharth** [comment]
@ibacher I guess this works perfectly alright

**Comment by ibacher** [comment]
So do we need this?

**Comment by VaishSiddharth** [comment]
@ibacher I tried the flow without this if statement and it worked. I think that there is some sort of order which needs to be followed while loading the modules. I mean that the FHIR2 module should be present to allow the smartonfhir module work properly otherwise it doesn't starts automatically.

Also I think that keeping those line might give us an extra check that the conformance urls are not restricted by authentication. But I am not very sure. It your take :D

**Comment by ibacher** [comment]
@VaishSiddharth Yeah, I merged this because I realised we probably want to expose those URLs with or without the SMART-on-FHIR module.

**Pull request merged**